src/parser/parser_env.ml

```diff
@@ -38,51 +38,32 @@ end
  * - Refactor this to memoize all requested lookahead, so we aren't lexing the
  *   same token multiple times.
  *)
-let maximum_lookahead = 2
 
 module Lookahead : sig
```

Comment:
Changes to this module look good! Aside from better perf, the simplicity makes this logic quite a bit easier to follow as well.